hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -721,20 +720,20 @@ private void initializeFileSystem() throws IOException {
   }
 
   private TableDescriptors getFsTableDescriptors() throws IOException {
-    return new FSTableDescriptors(this.conf,
-      this.dataFs, this.dataRootDir, !canUpdateTableDescriptor(), false, getMetaTableObserver());
-  }
-
-  protected Function<TableDescriptorBuilder, TableDescriptorBuilder> getMetaTableObserver() {
-    return null;
+    if (canUpdateTableDescriptor()) {
+      return new FSTableDescriptors(this.conf, this.dataFs, this.dataRootDir, false,
+        builder -> builder.setRegionReplication(
```

Comment:
This changes the semantic? Not sure why but before this patch we only set this for HMaster?